# 모두의 딥러닝 개정 3판
## 2장 딥러닝의 핵심 미리보기

6개의 feature \
17번째는 target

In [1]:
# 1. 환경 준비
from tensorflow.keras import models, layers
import numpy as np
import pandas as pd

In [8]:
df = pd.read_csv('data/ThoraricSurgery3.csv', header=None)
df

# df.head() row 5개만 보여주기

0     1     2   3   4   5   6   7   8   9   10  11  12  13  14  15  16
0     1  2.88  2.16   1   0   0   0   1   1   3   0   0   0   1   0  60   0
1     2  3.40  1.88   0   0   0   0   0   0   1   0   0   0   1   0  51   0
2     2  2.76  2.08   1   0   0   0   1   0   0   0   0   0   1   0  59   0
3     2  3.68  3.04   0   0   0   0   0   0   0   0   0   0   0   0  54   0
4     2  2.44  0.96   2   0   1   0   1   1   0   0   0   0   1   0  73   1
..   ..   ...   ...  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..
465   1  3.88  2.12   1   0   0   0   1   0   2   0   0   0   1   0  63   0
466   2  3.76  3.12   0   0   0   0   0   0   0   0   0   0   1   0  61   0
467   2  3.04  2.08   1   0   0   0   1   0   2   0   0   0   0   0  52   0
468   2  1.96  1.68   1   0   0   0   1   1   1   0   0   0   1   0  79   0
469   2  4.72  3.56   0   0   0   0   0   0   1   0   0   0   1   0  51   0

[470 rows x 17 columns]

- 샘플 수(환자 수): 470명
- 속성(의료 기록): 16가지
- 클래스(사망: 0 / 생존: 1)

In [ ]:
# 2. 데이터 준비
data_set = np.loadtxt('data/ThoraricSurgery3.csv', delimiter=',')  # delimiter:구분자
X = data_set[:, :16]  # : 모든 행  # 입력feature
y = data_set[:, 16]   # target

SubClassing : 하나에서 분화되었다가 다시 합처짐
Sequesntial Model : 층층이 쌓인 경우

Keras에서 제공하는 Fully Connected Layers
dense(출력 갯수, 입력 갯수, 활성화함수)
input 16 / hidden 30

앞쪽의 출력이 뒷쪽의 입력이므로 dense(출력, 입력생략, 활성화함수)

입력층 weight 16 30
입력층 bias 1 30

은닉층 weight 30 1
은닉층 bias 1

출력 param 541 (480+31)

In [ ]:
# 3. 구조 결정
model = models.Sequential()
model.add(layers.Dense(30, input_dim=16, activation='relu'))
model.add(layers.Dense(1, activation='sigmoid'))

In [ ]:
model.summary()

In [ ]:
# 4. 모델실행

# Keras에서 제공하는 이진분류 손실함수
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])

history = model.fit(X, y, epochs=5, batch_size=16)
# X(입력feature)가 y(target)임을 학습시키기
# 한 Epoch에 파라미터 갱신 30번